### PR TITLE
Suppress a possible relative_to() failure, fixes #9894

### LINKED ISSUE #9894: pathlib error when using python from double-slash //path
- [x] a detailed description of the bug or problem you are having

At our workplace path is usually written starting double-slash, like //path, to have compatibility between Linux and other OS platforms. We find recent versions of pytest does not work well when python's path starts //.  My understanding is that pathlib's relative_to() method is used somewhere under ExceptionInfo.getexpr() when it tries to process traceback entries, and pathlib's relative_to() does not work well when mixing single-slash vs double-slash paths. 

- [x] output of `pip list` from the virtual environment you are using

```
Package       Version
------------- -------
attrs         21.4.0
iniconfig     1.1.1
packaging     21.3
pip           20.3.4
pkg-resources 0.0.0
pluggy        1.0.0
py            1.11.0
pyparsing     3.0.8
pytest        7.1.2
setuptools    44.1.1
tomli         2.0.1
```

- [x] pytest and operating system versions

The issue exists in pytest >=7, on Linux. pytest < 7 is good.

- [x] minimal example if possible

```
$ cat test_foo.py
import pytest

# this will cause error
import nonexist

def test_foo():
    pass

# create venv from // python path
$ //usr/bin/python3 -m venv .venv-doubleslash

$ . .venv-doubleslash/bin/activate

(.venv-doubleslash) $ python -c "import sys;print(sys.base_prefix)"
//usr

(.venv-doubleslash) $ pip install pytest

(.venv-doubleslash) $ pytest .
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.1.2, pluggy-1.0.0
rootdir: /var/tmp/pytest-issue
collected 0 items                                                              
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/main.py", line 268, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/main.py", line 321, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/main.py", line 332, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/main.py", line 657, in perform_collect
INTERNALERROR>     self.items.extend(self.genitems(node))
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/main.py", line 824, in genitems
INTERNALERROR>     rep = collect_one_node(node)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/runner.py", line 537, in collect_one_node
INTERNALERROR>     rep: CollectReport = ihook.pytest_make_collect_report(collector=collector)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_callers.py", line 55, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/capture.py", line 770, in pytest_make_collect_report
INTERNALERROR>     rep = outcome.get_result()
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/runner.py", line 388, in pytest_make_collect_report
INTERNALERROR>     errorinfo = collector.repr_failure(call.excinfo)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/nodes.py", line 549, in repr_failure
INTERNALERROR>     return self._repr_failure_py(excinfo, style=tbstyle)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/nodes.py", line 475, in _repr_failure_py
INTERNALERROR>     return excinfo.getrepr(
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/_code/code.py", line 666, in getrepr
INTERNALERROR>     return fmt.repr_excinfo(self)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/_code/code.py", line 926, in repr_excinfo
INTERNALERROR>     reprtraceback = self.repr_traceback(excinfo_)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/_code/code.py", line 867, in repr_traceback
INTERNALERROR>     reprentry = self.repr_traceback_entry(entry, einfo)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/_code/code.py", line 825, in repr_traceback_entry
INTERNALERROR>     path = self._makepath(entry_path)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/_code/code.py", line 841, in _makepath
INTERNALERROR>     np = bestrelpath(Path.cwd(), path)
INTERNALERROR>   File "/var/tmp/pytest-issue/.venv-doubleslash/lib/python3.9/site-packages/_pytest/pathlib.py", line 711, in bestrelpath
INTERNALERROR>     reldest = dest.relative_to(base)
INTERNALERROR>   File "//usr/lib/python3.9/pathlib.py", line 928, in relative_to
INTERNALERROR>     raise ValueError("{!r} is not in the subpath of {!r}"
INTERNALERROR> ValueError: '//usr/lib/python3.9/pathlib.py' is not in the subpath of '/' OR one path is relative and the other is absolute.

============================ no tests ran in 0.02s =============================
```

### PR DESCRIPTION
**Pull request by stphnlyd**

### CODE REVIEW

File: src/_pytest/pathlib.py [hunk 1]
```diff
@@ -707,8 +707,12 @@ def bestrelpath(directory: Path, dest: Path) -> str:
     # Can be the case for a relative path and an absolute path.
     if not base:
         return str(dest)
-    reldirectory = directory.relative_to(base)
-    reldest = dest.relative_to(base)
+    try:
+        reldirectory = directory.relative_to(base)
+        reldest = dest.relative_to(base)
+    except ValueError:
```
Comment (nicoddemus) [inline src/_pytest/pathlib.py:713]: Could you please add a test for `bestrelpath` which tests this condition? It is important to avoid future regressions.

Comment (stphnlyd) [reply src/_pytest/pathlib.py:713]: Absolutely. Just added and let's wait for the checks to finish..

File: testing/test_pathlib.py [hunk 2]
```diff
@@ -413,6 +413,11 @@ def test_bestrelpath() -> None:
     assert bestrelpath(curdir, curdir.parent) == ".."
     assert bestrelpath(curdir, Path("hello")) == "hello"
 
+    # GH 9894
+    directory = Path("/var/tmp/pytest-issue")
```
Comment (nicoddemus) [inline testing/test_pathlib.py:417]: This test passes for me on Windows (which is fine), but @stphnlyd did you double check that this check fails without the rest of your changes?

File: src/_pytest/pathlib.py (hunk 1 above)
Comment (nicoddemus) [reply src/_pytest/pathlib.py:713]: Thanks!

File: testing/test_pathlib.py (hunk 2 above)
Comment (stphnlyd) [reply testing/test_pathlib.py:417]: Yes with the change in pathlib.py both this test and my demo case in #9894 fail.

**Review by asottile — COMMENTED** [review]
I really don't think pytest should be making a change here -- we're really just codifying this upstream bug here (it should be able to make a relative path -- we shouldn't be producing an absolute path from this function)

**Comment by nicoddemus** [comment]
Hmm I will defer this to @asottile then. 👍 

Feel free to close this PR and the issue if you think that's the correct approach.

**Pull request closed without merging**